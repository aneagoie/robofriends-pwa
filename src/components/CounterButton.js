import React, { Component } from "react";

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  render() {
    return (
      <button
        id="counter"
        color={this.props.color}
        onClick={() => this.setState({ count: this.state.count + 1 })}
      >
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
