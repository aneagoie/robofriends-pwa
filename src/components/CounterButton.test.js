import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import CounterButton from "./CounterButton"
import userEvent from '@testing-library/user-event';

it('renders the CounterButton component correctly', () => {
    const mockColor = 'red'
    const { asFragment } = render(<CounterButton  color={mockColor} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("correctly increments the counter", async() => {
    const mockColor = 'red'
    render(<CounterButton color={mockColor} />);
    const counterButton = screen.getByTestId('counter');
    expect(counterButton).toBeInTheDocument();
    await userEvent.click(counterButton);
    await userEvent.click(counterButton);
    expect(counterButton.textContent).toBe('Count: 3'); // because initial count is 1
    // Simulate a generic keyboard event (e.g., pressing the "a" key)
    await userEvent.keyboard('a');
    // The counter should remain the same if the CounterButton does not handle keyboard events
    expect(counterButton.textContent).toBe('Count: 3');
    // Instead of checking the prop, check the rendered button's style
    expect(counterButton).toHaveAttribute('color', mockColor);
  });
