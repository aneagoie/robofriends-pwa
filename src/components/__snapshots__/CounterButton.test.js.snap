// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CounterButton component correctly 1`] = `
<DocumentFragment>
  <button
    color="red"
    data-testid="counter"
  >
    Count: 1
  </button>
</DocumentFragment>
`;
